wrapanapi/systems/redfish.py

```diff
@@ -33,5 +130,61 @@ def _identifying_attrs(self):
     def info(self):
         return 'RedfishSystem url={}'.format(self.url)
 
-    def __del__(self):
-        """Disconnect from the API when the object is deleted"""
+    def server_stats(self, *requested_stats, **kwargs):
+        """
+        Evaluate the requested server stats at the API server.
+
+        Returns a dictionary of stats and their respective evaluated values.
+
+        Args:
+            requester: the physical server instance to check the stats on
+            requested_stats: a list of named stats to be checked
+        """
+        # Get the requester which represents the class of this method's caller
+        requester = kwargs.get('requester')
+
+        # Retrieve and return the stats
+        requested_stats = requested_stats or self._stats_available
+
+        # Get an instance of the requested Redfish server
+        redfish_server = self.get_server(requester.ems_ref)
+
+        return {stat: self._server_stats_available[stat](redfish_server)
+                for stat in requested_stats}
+
+    def server_inventory(self, *requested_items, **kwargs):
+        """
+        Evaluate the requested inventory item statuses at the API server.
+
+        Returns a dictionary of items and their respective evaluated values.
+
+        Args:
+            requester: the physical server instance to check the stats on
+            requested_items: a list of named items to be checked
+        """
+        # Get the requester which represents the class of this method's caller
+        requester = kwargs.get('requester')
```

Comment:
This appears to be required, as you later reference an attribute on `requester`. 

I don't see the need to take `kwargs` here, what's the reasoning behind this kwarg lookup instead of just a required `requester` parameter listed before `*requested_items`?

Reply:
Doing that doesn't work because of the way this method is called.
But please see my PRs https://github.com/ManageIQ/wrapanapi/pull/331 and https://github.com/ManageIQ/integration_tests/pull/7735 where I address this issue.

Reply:
This PR is for new Redfish-specific features. PR #331 and ManageIQ/integration_tests#7735 are changing a signature that results changes both in Lenovo's code and the Redfish provider. And the genesis of the PRs was really 1) add features to Redfish provider by using the Lenovo's code as example, and 2) address comments that required a deeper refactor later.

If need be, I can reorder the PRs, or I can put those commits here, but that will really muddy the purpose.

Reply:
Apologies, I didn't answer the last question: while this is a new method, it is called from [cfme/physical/physical_server.py](
https://github.com/ManageIQ/integration_tests/blob/2c39a4e47ca25c7e9e97e6ee3ab9d796b76a3b24/cfme/physical/physical_server.py#L240-L241), i.e. a piece of code in the integration tests that is common to all physical servers - Lenovo's xClarity, Redfish.